pvlib/modelchain.py

```diff
@@ -776,24 +887,30 @@ def infer_aoi_model(self):
                              'kwarg; or set aoi_model="no_loss".')
 
     def ashrae_aoi_loss(self):
-        self.aoi_modifier = self.system.get_iam(self.aoi, iam_model='ashrae')
+        self.results.aoi_modifier = self.system.get_iam(
+            self.results.aoi, iam_model='ashrae')
         return self
 
     def physical_aoi_loss(self):
-        self.aoi_modifier = self.system.get_iam(self.aoi, iam_model='physical')
+        self.results.aoi_modifier = self.system.get_iam(self.results.aoi,
+                                                        iam_model='physical')
         return self
 
     def sapm_aoi_loss(self):
-        self.aoi_modifier = self.system.get_iam(self.aoi, iam_model='sapm')
+        self.results.aoi_modifier = self.system.get_iam(self.results.aoi,
+                                                        iam_model='sapm')
         return self
 
     def martin_ruiz_aoi_loss(self):
-        self.aoi_modifier = self.system.get_iam(self.aoi,
+        self.results.aoi_modifier = self.system.get_iam(self.results.aoi,
                                                 iam_model='martin_ruiz')
         return self
 
     def no_aoi_loss(self):
-        self.aoi_modifier = 1.0
+        if self.system.num_arrays == 1:
+            self.results.aoi_modifier = 1.0
+        else:
+            self.results.aoi_modifier = (1.0,) * self.system.num_arrays
```

Comment:
I'm not sure we need this case. We already use a singleton instead `pd.Series(1.0, index=weather.index)`.

Reply:
I don't totally follow, but we do actually need this so that `self.results.aoi_modifier` is the correct type. Since this is a "per-array" quantity, it is required to be a tuple when the number of arrays is > 1. If it is not a tuple `ModelChain.effective_irradiance_model()` will fail.

Reply:
Understood on the downstream failure. My point here is that while aoi modifier is nominally a per time quantity but we still assign only the scalar 1.0 for no loss, and similarly perhaps for no loss we should use the single scalar for all of the arrays. I don't have a strong feeling about this though.

Reply:
If we want to do that, I think the appropriate place would be in `PVSystem`. I'm not really keen on it since it adds complexity with (I think) very little benefit.